zfs_snapshot.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+NOW=$(date --iso-8601=minutes | sed 's/:/-/g' | cut -c 1-10)
+DELETE_OLDEST=$(date -d "$BACKUP_TIME ago" --iso-8601=minutes | sed 's/:/-/g' | cut -c 1-10)
+LAST_SNAPSHOT=$(zfs list -t snapshot | tail -n1 | cut -d " " -f 1)
+
+# Switch remote PostgreSQL to backup mode, take a ZFS snapshot, and exit from PostgreSQL backup mode
+
+psql --host=localhost --username=postgres -c  "SELECT pg_start_backup('$NOW', true);"
+sudo zfs snapshot large-storage-pool/data@$NOW
+psql --host=localhost --username=postgres -c  "SELECT pg_stop_backup();"
+
+# Destroy oldest snapshot on local server
+sudo zfs destroy -v large-storage-pool/data@$DELETE_OLDEST && zfs destroy -v large-storage-pool/data@$DELETE_OLDEST
+
+# Copy newly taken snapshot from remote server
+
+sudo zfs send -cRi $LAST_SNAPSHOT large-storage-pool/data@$NOW" | ssh $BACKUP_SERVER sudo zfs receive -vF $BACKUP_DIRECTORY/data/data@$NOW
```

Comment:
Can you make backup server user and SSH key explicit for this command, so they are also given as an input? This may the script is more deterministic and you can see all inputs that need to go in  - the script fails if you do not give it the correct SSH key.